components/core/src/clp/streaming_archive/single_file_archive/writer.cpp

```diff
@@ -0,0 +1,297 @@
+#include "writer.hpp"
+
+#include <cstddef>
+#include <cstring>
+#include <filesystem>
+#include <sstream>
+#include <vector>
+
+#include <fmt/core.h>
+#include <msgpack.hpp>
+#include <spdlog.h>
+
+#include "../../Defs.h"
+#include "../../ErrorCode.hpp"
+#include "../../FileReader.hpp"
+#include "../../FileWriter.hpp"
+#include "../../TraceableException.hpp"
+#include "../ArchiveMetadata.hpp"
+#include "../Constants.hpp"
+#include "Defs.hpp"
+
+namespace clp::streaming_archive::single_file_archive {
+
+namespace {
+constexpr size_t cReadBlockSize = 4096;
+
+/**
+ * @param next_segment_id ID of the next segment to be created in the archive.
+ * @return Vector of segment IDs.
+ */
+[[nodiscard]] auto get_segment_ids(segment_id_t next_segment_id) -> std::vector<std::string>;
```

Comment:
imo, it's weird to store the seg_ids as strings since we know they are int. can we stored them as segment_id_t and convert to string later when we need to use string?